lib/modules/manager/npm/post-update/npm.ts

```diff
@@ -32,8 +32,9 @@ import { getPackageManagerVersion, lazyLoadPackageJson } from './utils';
 
 async function getNpmConstraintFromPackageLock(
   lockFileDir: string,
+  lockFileName: string,
 ): Promise<string | null> {
-  const packageLockFileName = upath.join(lockFileDir, 'package-lock.json');
+  const packageLockFileName = upath.join(lockFileDir, lockFileName);
```

Comment:
I think lockFileName maybe the full path, so this would double up the subdirectory if it was e.g. foo/package-lock.json. I think we might be able to replace lockFileDir with lockFileName in this function signature

Reply:
Renamed `lockFileName` to `filename` to be consitent with the rest of the file, thanks!

Reply:
I prefer lockFileName, and it doesn't fix the problem I identified

Reply:
Than it means that https://github.com/renovatebot/renovate/pull/25820 also won't in the case you've mentioned, right? I struggle to find what is the issue.

Reply:
It works. Think if lockFileName=foo/package-lock.json. Then lockFileDir=foo. If you join them together then you get foo/foo/package-lock.json

Reply:
Okay, that's why I've renamed the variable cause it was confusing. In the `npm.ts` file variable `lockFileName` is used as full path (like you've mentioned) but in this particular version both `lockFileDir` and `filename` (which will be either `package-lock.json` or `npm-shrinkwrap.json`) are passed. The full path is constructed in https://github.com/renovatebot/renovate/pull/26726/files#diff-004a30b9b53101a638c5deba63490af1be69ddb86e06750c1ff7e786c1640c85R37 like in https://github.com/renovatebot/renovate/pull/26726/files#diff-004a30b9b53101a638c5deba63490af1be69ddb86e06750c1ff7e786c1640c85R70

Those to are constrructed in separate functions, and just because in the latter `lockFileName` indicates full path, I've renamed the argument name to be consitent. 
The only difference between your solution and mine is that we are only reading/checking additional file - `npm-shrinkwrap.json` (in case `package-lock.json` is missing) - all the rest remains as it was in https://github.com/renovatebot/renovate/pull/25820